pkg/configprovider/globprovider/provider.go

```diff
@@ -22,27 +22,33 @@ import (
 	"sort"
 	"strings"
 
-	"gopkg.in/yaml.v3"
-
+	"github.com/SumoLogic/sumologic-otel-collector/pkg/configprovider/providerutil"
 	"go.opentelemetry.io/collector/confmap"
+	"gopkg.in/yaml.v3"
 )
 
 const (
 	schemeName   = "glob"
 	schemePrefix = schemeName + ":"
 )
 
-type provider struct{}
+type Provider struct{
```

Comment:
Why do we need this to be public?

Reply:
https://github.com/SumoLogic/sumologic-otel-collector/blob/j_config_merge_override/pkg/configprovider/opampprovider/provider.go#L88

To do the type casting here.